Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -712,24 +715,81 @@ void WiiSockMan::Update()
       socket_iter = WiiSockets.erase(socket_iter);
     }
   }
-  s32 ret = select(nfds, &read_fds, &write_fds, &except_fds, &t);
+  s32 ret = poll(ufds.data(), static_cast<u32>(ufds.size()), 0);
+  if (ret < 0)
+  {
+    ERROR_LOG(IOS_NET, "SO_POLL poll failed (%d)", GetNetErrorCode(ret, "Update", true));
+    return;
+  }
 
-  if (ret >= 0)
+  int n = 0;
+  for (auto& elem : WiiSockets)
   {
-    for (auto& pair : WiiSockets)
-    {
-      WiiSocket& sock = pair.second;
-      sock.Update(FD_ISSET(sock.fd, &read_fds) != 0, FD_ISSET(sock.fd, &write_fds) != 0,
-                  FD_ISSET(sock.fd, &except_fds) != 0);
-    }
+    elem.second.Update(!!(ufds[n].revents & read_event), !!(ufds[n].revents & write_event),
+                       !!(ufds[n].revents & error_event));
+    n++;
   }
-  else
+
+  UpdatePollCommands();
+}
+
+void WiiSockMan::UpdatePollCommands()
+{
+  constexpr static int error_event = (POLLHUP | POLLERR);
+
+  if (pending_polls.empty())
+    return;
+
+  const std::chrono::time_point<std::chrono::high_resolution_clock> now =
```

Comment:
Auto is particularly appropriate here.

```suggestion
  const auto now =
```